### ISSUE TITLE
[Module Proposal]: `avm/ptn/authorization/role-definition`

### ISSUE THREAD

**Issue by jtracey93** [Azure/Azure-Verified-Modules#1757]
### Check for previous/existing GitHub issues/module proposals

- [x] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [x] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Bicep

### Module Classification?

Pattern Module

### Module Name

avm/ptn/authorization/role-definition

### Module Details

This module is required for the ALZ Bicep refactor to AVM.

Initially the module will only support role definition creation at the Management Group scope, but can be expanded later if required to additional scopes, like the policy assignment pattern module etc. (but not required for ALZ Bicep so out of scope for now, for speed)

cc: @oZakari 

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

jtracey93

### (Optional) Secondary Module Owner's GitHub Username (handle)

ozakari

**Comment by jtracey93** [comment]
I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER

**Comment by matebarabas** [comment]
Hi @jtracey93,

Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

Before starting development, please ensure ALL the following requirements are met.

**Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:

- For your module:
  - `ModuleName` - for naming your module
  - `TelemetryIdPrefix` - for your module's [telemetry](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr3---category-telemetry---deploymentusage-telemetry)
- For your module's repository:
  - Repo name and folder path are defined in `RepoURL`
  - Create GitHub teams for module owners and contributors and grant them permissions as outlined [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only).
  - Grant permissions for the AVM core team and PG teams on your GitHub repo as described [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions).

Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍

You can now start the development of this module! ✅ Happy coding! 🎉

**Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team

**Comment by jtracey93** [comment]
@matebarabas PR ready to go https://github.com/Azure/bicep-registry-modules/pull/3931

However, test failing as CSV not updated, let me know when it is and i can re-rerun tests :)

**Comment by matebarabas** [comment]
CSV updated; MAR update PR submitted just this morning.

**Comment by jtracey93** [comment]
@matebarabas this is now published 👍

**Comment by matebarabas** [comment]
- [x] Module published.
- [x] GitHub teams are available and assigned to their parents.
- [x] `CODEOWNERS` file updated.
- [x] Issue template file updated.

The module index will be updated to reflect this change.
 
Thank you for your work @jtracey93!
